apps/src/gamelab/GameLab.js

```diff
@@ -408,6 +443,46 @@ GameLab.prototype.runButtonClick = function () {
   this.execute();
 };
 
+function p5KeyCodeFromArrow(idBtn) {
+  switch (idBtn) {
+    case ArrowIds.LEFT:
+      return window.p5.prototype.LEFT_ARROW;
+    case ArrowIds.RIGHT:
+      return window.p5.prototype.RIGHT_ARROW;
+    case ArrowIds.UP:
+      return window.p5.prototype.UP_ARROW;
+    case ArrowIds.DOWN:
+      return window.p5.prototype.DOWN_ARROW;
+  }
+}
+
+GameLab.prototype.onArrowButtonDown = function (buttonId, e) {
+  // Store the most recent event type per-button
+  this.btnState[buttonId] = ButtonState.DOWN;
+  e.preventDefault();  // Stop normal events so we see mouseup later.
+
+  this.gameLabP5.notifyKeyCodeDown(p5KeyCodeFromArrow(buttonId));
+};
+
+GameLab.prototype.onArrowButtonUp = function (buttonId, e) {
+  // Store the most recent event type per-button
+  this.btnState[buttonId] = ButtonState.UP;
+
+  this.gameLabP5.notifyKeyCodeUp(p5KeyCodeFromArrow(buttonId));
+};
+
+GameLab.prototype.onMouseUp = function (e) {
+  // Reset all arrow buttons on "global mouse up" - this handles the case where
+  // the mouse moved off the arrow button and was released somewhere else
```

Comment:
We'll have to try it out. It does maintain state per-button, so if mouse events are sent simultaneously, it should work. Or multi-touch scenarios may require you to actually look at the touch events (not the mouse events)... But this is more of a question for playlab, since gamelab will move away from this UI metaphor pretty quickly...